### ISSUE TITLE
Remove boilerplate repository methods.

### ISSUE THREAD

**Issue by psycho-ir** [zalando/java-sproc-wrapper#15]
What do you think about removing AbstractSProcService subclasses which need to implement all SProcService methods?
I think we can use spring-data to generate SP wrappers dynamically.
If you think it worth, i can start!

**Comment by Jan-M** [comment]
I am not 100% sure what and why you want to remove it, we have some projects using the 

T SProcProxyBuilder.build(ds, T);

method directly to avoid implementing anything else if i remember correctly.

Any change to make integration with Spring less work is a good idea though, originally we had some use cases for this layer, but they are basically rare.

**Comment by psycho-ir** [comment]
My idea is just to declare required interfaces, something like spring-data-jpa.
So for every Stored Procedure Service, Developer needs just to declare method names and use some annotations and in runtime the stored procedures are called by proxies on the fly.

I thought this approach maybe reduce the effort of development.

**Comment by mkdika** [comment]
@psycho-ir I agree with you, from the first place I though that it is something like declare Spring Data JPA's Repository interface, then auto-magically :) and I realized this project have been almost a year from its last update, waiting for update.

**Comment by vadeg** [comment]
@mkdika It was updated to Spring 4 last year :) Maybe it is worth no to implement JPA-like solution but extend existing Spring JPA implementation.
